embulk-input-s3/src/main/java/org/embulk/input/s3/AbstractS3FileInputPlugin.java

```diff
@@ -67,6 +67,11 @@
         @ConfigDefault("null")
         public Optional<String> getAccessKeyId();
 
+        @Config("http_proxy")
+        @ConfigDefault("null")
+        public Optional<HttpProxy> getHttpProxy();
+        public void setHttpProxy(Optional<HttpProxy> httpProxy);
```

Comment:
I don't think so to be honest. If users are typing their configurations every runtime, it may be useful to skip typing such `env` configurations. But, it's just per making configurations. Does this really matter? Users can utilize Liquid templates as well.

I agree that the benefits are not zero, but I think the risk is larger by that default proxy configuration.

* While we encourage to use this `HttpProxy` class for many plugins, but our encourage will not cover all. `HttpProxy` is just a tooling. Implicit configurations may cause confusions especially when used with third-party non-`HttpProxy` plugins.
* For your latter case, there are expectedly many cases as well that no proxy is used for input while output needs proxy: such as uploading (internal) application server data into external services. The risk of later confusion is expectedly larger than just a few key types in making a configuration.